lib/appsignal/transaction.rb

```diff
@@ -583,6 +623,18 @@ def sample_data
       end
     end
 
+    def duplicate
+      new_transaction_id = SecureRandom.uuid
+
+      self.class.new(
+        new_transaction_id,
+        namespace,
+        :ext => ext.duplicate(new_transaction_id)
```

Comment:
I'd rather generate a new transaction ID for each duplicate. Any reason you want to reuse the same transaction ID?

(I'm even considering remove the transaction ID argument from the Transaction initializer.